google/google-how-to/gce/set-hostname-using-cloudinit.rst

```diff
@@ -0,0 +1,77 @@
+Set ``hostname`` using cloud-init
+=================================
+
+One way of setting your VM's ``hostname`` is to use a set-up tool like cloud-init. The process for setting it involves `using the gcloud add-metadata command`_ to program the cloud-init metadata. Use:
+
+.. code::
+
+    gcloud compute instances add-metadata INSTANCE_NAME --metadata-from-file=KEY=VALUE
+
+where INSTANCE_NAME is your VM name and the metadata is specified using a KEY=VALUE pair, e.g. user-data=FILENAME. Here FILENAME is the local path to a file that contains the desired cloud-init configurations, including the hostname:
+
+.. code:: 
+
+    hostname: test123
+
+This process of setting ``hostname`` using cloud-init requires slight changes based on the version of Ubuntu that your GCE image is using. Starting from Ubuntu 24.04, ``/etc/hostname`` is no longer present by default and the cloud-init key ``create_hostname_file`` is set to false. This means that the process mentioned above will not work as is and needs some tweaking. 
+
+
+Set ``hostname`` in Ubuntu 24.04 and later
+------------------------------------------
+
+Using cloud-init (24.04+)
```

Comment:
Sorry, I had added the extra version numbers in the brackets to avoid a duplicate label warning. I now realise the confusion that creates! Anyway, those are gone now.